datastore/integration_test.go

```diff
@@ -1432,62 +1425,51 @@ func createTestEntities(ctx context.Context, t *testing.T, client *Client, parti
 }
 
 type runWithOptionsTestcase struct {
-	desc         string
-	wantKeys     []*Key
-	wantStats    *ResultSetStats
-	wantEntities []SQChild
-	opts         []RunOption
+	desc               string
+	wantKeys           []*Key
+	wantExecutionStats *ExecutionStats
+	wantPlan           *Plan
+	wantEntities       []SQChild
+	opts               []RunOption
 }
 
 func getRunWithOptionsTestcases(ctx context.Context, t *testing.T, client *Client, partialNameKey string, count int) ([]runWithOptionsTestcase, int64, *Key, func()) {
 	keys, entities, now, parent, cleanup := createTestEntities(ctx, t, client, partialNameKey, count)
 	return []runWithOptionsTestcase{
 		{
-			desc:         "No mode",
+			desc:         "No ExplainOptions",
 			wantKeys:     keys,
 			wantEntities: entities,
 		},
 		{
-			desc:         "Normal query mode",
-			opts:         []RunOption{QueryModeNormal},
-			wantKeys:     keys,
-			wantEntities: entities,
-		},
-		{
-			desc: "Explain query mode",
-			opts: []RunOption{QueryModeExplain},
-			wantStats: &ResultSetStats{
-				QueryPlan: &QueryPlan{
-					PlanInfo: map[string]interface{}{
-						"indexes_used": []interface{}{
-							map[string]interface{}{
-								"properties":  "(T ASC, I ASC, __name__ ASC)",
-								"query_scope": "Includes Ancestors",
-							},
-						},
+			desc: "ExplainOptions.Analyze is false",
+			opts: []RunOption{ExplainOptions{}},
+			wantPlan: &Plan{
+				IndexesUsed: []*map[string]interface{}{
+					{
+						"properties":  "(T ASC, I ASC, __name__ ASC)",
+						"query_scope": "Includes Ancestors",
 					},
 				},
 			},
 		},
 		{
-			desc:     "ExplainAnalyze query mode",
-			opts:     []RunOption{QueryModeExplainAnalyze},
+			desc:     "ExplainAnalyze.Analyze is true",
+			opts:     []RunOption{ExplainOptions{Analyze: true}},
 			wantKeys: keys,
-			wantStats: &ResultSetStats{
-				QueryPlan: &QueryPlan{
-					PlanInfo: map[string]interface{}{
-						"indexes_used": []interface{}{
-							map[string]interface{}{
-								"properties":  "(T ASC, I ASC, __name__ ASC)",
-								"query_scope": "Includes Ancestors",
-							},
-						},
-					},
-				},
-				QueryStats: map[string]interface{}{
+			wantExecutionStats: &ExecutionStats{
```

Comment:
Maybe this is just a bi-product of the test environment, but do you know why this result doesn't contain bytesReturned, executionDuration and readOptions?

Reply:
New protos do not have bytesReturned.
For rest:
The environment does return those but they are unpredictable. So, they have not been added in expected results and are being ignored while comparing expected and actual.